### ISSUE TITLE
Add portScanRange for TCP/IP connectivity

### ISSUE THREAD

**Issue by TomaszGaweda** [hazelcast/hazelcast#16859]
Hi,

would be great to have option like portScanRange for TCP/IP join. The reason is, when there are few clusters in the network, TCP join scans 5701, 5702, 5703 and stops. We can increase `hazelcast.tcp.join.port.try.count` to higher numbers, but it unnecessarily scans many ports and makes joining longer. This is especially visible on Jenkins with many automated tests, but also may affect production systems with mutiple grid clusters.

We may also consider scanStartingPort, which can be used with `hazelcast.tcp.join.port.try.count` to create behaviour like `portScanRange` will give.

**Comment by mmedenjak** [comment]
Hi @TomaszGaweda ! I'll put this enhancement into 4.1 but I don't guarantee we'll find the time to implement it. Since I assume it will be rather simple, do you think you could send a PR? We have a new CLA system that is really simple and that you can try out :)

**Comment by TomaszGaweda** [comment]
Yes, I'll try to make this change today. Was busy with stuff from work so decided to create an issue first and implement when time allows :) CLA already signed for Jet's PR
